File: 2학년 2학기/기타/하노이 탑.py
def printHanoi(n, left, mid, right):
    if n == 1:
        print(f"{left} {right}")
    else:
        printHanoi(n-1, left, right, mid)
        print(f"{left} {right}")
        printHanoi(n-1, mid, left, right)

k = int(input())
print((1<<k)-1)
if k <= 20:
    printHanoi(k, 1, 2, 3)

File: 2019-2021/2학년 2학기/기타/하노이 탑.cpp
#include <iostream>

void printHanoi(int n, int left, int mid, int right) {
    if (n == 1) {
        printf("%d %d\n", left, right);
    } else {
        printHanoi(n-1, left, right, mid);
        printf("%d %d\n", left, right);
        printHanoi(n-1, mid, left, right);
    }
}

int main() {
    int k;
    scanf("%d", &k);

    printf("%d\n", (1<<k)-1);
    printHanoi(k, 1, 2, 3);

    return 0;
}